tests/integration/test_repos_branch.py

```diff
@@ -0,0 +1,56 @@
+"""Integration tests for methods implemented on Branch."""
+import github3
+
+from .helper import IntegrationHelper
+
+
+class TestBranch(IntegrationHelper):
+    """Branch integration tests."""
+
+    betamax_kwargs = {'match_requests_on': ['method', 'uri', 'json-body']}
+
+    def test_protect(self):
+        expected = {
+            'enabled': True,
+            'required_status_checks': {'enforcement_level': 'off',
+                                       'contexts': []}}
+        required = expected['required_status_checks']
+
+        self.token_login()
+        cassette_name = self.cassette_name('protect')
+        with self.recorder.use_cassette(cassette_name, **self.betamax_kwargs):
+            repository = self.gh.repository('bboe', 'github3.py')
```

Comment:
I was hoping I could write individual unit tests for each condition, but mocking out the the pieces is tedious and exactly what betamax makes simpler.